core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/services/dot-page-state/dot-page-state.service.ts

```diff
@@ -122,6 +122,29 @@ export class DotPageStateService {
      * @memberof DotPageStateService
      */
     setInternalNavigationState(state: DotPageRenderState): void {
+        const urlParam = generateDotFavoritePageUrl({
+            deviceInode: state.viewAs.device?.inode,
+            languageId: state.viewAs.language.id,
+            pageURI: state.page.pageURI,
+            siteId: state.site?.identifier
+        });
+
+        this.dotESContentService
+            .get({
+                itemsPerPage: 10,
+                offset: '0',
+                query: `+contentType:DotFavoritePage +deleted:false +working:true +owner:${state.user.userId} +DotFavoritePage.url_dotraw:${urlParam}`
```

Comment:
We need a dotFavPage service that encapsulate this, we keep doing this and if we need to change this query we have to touch too many places.

Reply:
done!